src/rust/ide/lib/view/src/graph_editor/builtin/visualization/native/documentation.rs

```diff
@@ -0,0 +1,208 @@
+        let screen          = scene.camera().screen();
+        let doc_view_height = screen.height - (DOC_VIEW_MARGIN * 2.0);
+        let size_vec        = Vector2(DOC_VIEW_WIDTH,doc_view_height);
+        let size            = Rc::new(Cell::new(size_vec));
+
+        dom.dom().set_style_or_warn("white-space"     ,"normal"                        ,&logger);
+        dom.dom().set_style_or_warn("overflow-y"      ,"auto"                          ,&logger);
+        dom.dom().set_style_or_warn("overflow-x"      ,"auto"                          ,&logger);
+        dom.dom().set_style_or_warn("background-color","rgba(255, 255, 255, 0.85)"     ,&logger);
+        dom.dom().set_style_or_warn("padding"         ,"5px"                           ,&logger);
+        dom.dom().set_style_or_warn("pointer-events"  ,"auto"                          ,&logger);
+        dom.dom().set_style_or_warn("border-radius"   ,format!("{}px", CORNER_RADIUS)  ,&logger);
+        dom.dom().set_style_or_warn("width"           ,format!("{}px", DOC_VIEW_WIDTH) ,&logger);
+        dom.dom().set_style_or_warn("height"          ,format!("{}px", doc_view_height),&logger);
+
+        scene.dom.layers.main.manage(&dom);
+        ViewModel {dom,logger,size}.init()
+    }
+
+    fn init(self) -> Self {
+        self.reload_style();
+        self
+    }
+
+    /// Sets size of the documentation view.
+    fn set_size(&self, size:Vector2) {
+        self.size.set(size);
+        self.reload_style();
+    }
+
+    /// Generates HTML documentation from documented Enso code.
+    fn gen_html_from(program:String) -> FallibleResult<String> {
+        let parser = parser::DocParser::new()?;
+        let output = parser.generate_html_docs(program);
+        Ok(output?)
+    }
+
+    /// Generates HTML documentation from pure Enso documentation.
+    fn gen_html_from_pure(doc:String) -> FallibleResult<String> {
+        let parser = parser::DocParser::new()?;
+        let output = parser.generate_html_doc_pure(doc);
+        Ok(output?)
+    }
+
+    /// Prepares data string for Doc Parser to work with after getting deserialization.
+    /// FIXME : Removes characters that are not supported by Doc Parser yet.
+    ///         https://github.com/enso-org/enso/issues/1063
+    fn prepare_data_string(data_inner:&visualization::Json) -> String {
+        let data_str = serde_json::to_string_pretty(&**data_inner);
+        let data_str = data_str.unwrap_or_else(|e| format!("<Cannot render data: {}>", e));
+        let data_str = data_str.replace("\\n", "\n");
+        data_str.replace("\"", "")
+    }
+
+    /// Creates a container for generated content and embeds it with stylesheet.
+    fn push_to_dom(&self, content:String) {
+        let data_str = format!(r#"<div class="docVis">{}{}</div>"#, doc_style(), content);
+        self.dom.dom().set_inner_html(&data_str)
+    }
+
+    /// Receives data, processes and presents it in the documentation view.
```

Comment:
All documentaiton sentences should use the form of `Receive data, process ...` not `ReceiveS data, processES, ...`. This is official rust comment guid.e I was writing about it in one of the previous reviews already.